src/dune_rules/pkg_rules.ml

```diff
@@ -193,7 +195,114 @@ module Value_list_env = struct
   let add_path (t : t) var path : t =
     Env.Map.update t var ~f:(fun paths ->
       let paths = Option.value paths ~default:[] in
-      Some (Value.Dir (Path.build path) :: paths))
+      Some (Value.Dir path :: paths))
+  ;;
+
+  (* Prepends the bin dir of a given toolchain compiler to the PATH
+     variable in the environment. *)
+  let add_toolchain_bin_dir_to_path t compiler =
+    let toolchain_bin_dir =
+      Toolchain.Compiler.bin_dir compiler |> Path.outside_build_dir
+    in
+    add_path t Env_path.var toolchain_bin_dir
+  ;;
+end
+
+module DB = struct
+  type t =
+    { all : Lock_dir.Pkg.t Package.Name.Map.t
+    ; system_provided : Package.Name.Set.t
+    ; available_compilers : Toolchain.Available_compilers.t
```

Comment:
Why do we need `available_compilers` here ? We are loading this from `opam-repository` when it is not used.

Reply:
Good point, I've removed it from the DB. It's a better approach than to make it an `option` as then the semantics of `DB.equal` are not dependent on `use_toolchains`.